### ISSUE TITLE
typescript fails when trying to build current master

### ISSUE THREAD

**Issue by matthias-k** [jupyter/nbdime#364]
Hi,

I am trying to install nbdime from github. However, the installation fails, apparently because typescript finds a lot of errors. Here is the start of the errors in the output:

```
    [...]
    > lerna run build                                                                                                                               [857/9418]
    
    lerna info version 2.10.1
    lerna info versioning independent
    lerna ERR! build Errored while running script in 'nbdime'
    lerna ERR! execute callback with error
    lerna ERR! Error: Command failed: npm run build
    lerna ERR! npm WARN invalid config loglevel="notice"
    lerna ERR!
    lerna ERR! npm ERR! Linux 4.4.0-116-generic
    lerna ERR! npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "run" "build"
    lerna ERR! npm ERR! node v8.10.0
    lerna ERR! npm ERR! npm  v3.5.2
    lerna ERR! npm ERR! code ELIFECYCLE
    lerna ERR! npm ERR! nbdime@0.4.0 build: `tsc --project src && node scripts/copy-files.js`
    lerna ERR! npm ERR! Exit status 1
    lerna ERR! npm ERR!
    lerna ERR! npm ERR! Failed at the nbdime@0.4.0 build script 'tsc --project src && node scripts/copy-files.js'.
    lerna ERR! npm ERR! Make sure you have the latest version of node.js and npm installed.
    lerna ERR! npm ERR! If you do, this is most likely a problem with the nbdime package,
    lerna ERR! npm ERR! not with npm itself.
    lerna ERR! npm ERR! Tell the author that this fails on your system:
    lerna ERR! npm ERR!     tsc --project src && node scripts/copy-files.js
    lerna ERR! npm ERR! You can get information on how to open an issue for this project with:
    lerna ERR! npm ERR!     npm bugs nbdime
    lerna ERR! npm ERR! Or if that isn't available, you can get their info via:
    lerna ERR! npm ERR!     npm owner ls nbdime
    lerna ERR! npm ERR! There is likely additional logging output above.
    lerna ERR!
    lerna ERR! npm ERR! Please include the following file with any support request:
    lerna ERR! npm ERR!     /tmp/nbdime/packages/nbdime/npm-debug.log
    lerna ERR!
    lerna ERR! > nbdime@0.4.0 build /tmp/nbdime/packages/nbdime
    lerna ERR! > tsc --project src && node scripts/copy-files.js
    lerna ERR!
    lerna ERR! src/common/collapsiblepanel.ts(7,8): error TS2307: Cannot find module '@phosphor/widgets'.
    lerna ERR! src/common/collapsiblepanel.ts(41,10): error TS2339: Property 'addClass' does not exist on type 'CollapsiblePanel'.
    lerna ERR! src/common/collapsiblepanel.ts(49,10): error TS2339: Property 'addWidget' does not exist on type 'CollapsiblePanel'.
    lerna ERR! src/common/collapsiblepanel.ts(56,10): error TS2339: Property 'addWidget' does not exist on type 'CollapsiblePanel'.
    lerna ERR! src/common/dragpanel.ts(7,8): error TS2307: Cannot find module '@phosphor/widgets'.
    lerna ERR! src/common/dragpanel.ts(11,8): error TS2307: Cannot find module '@phosphor/messaging'.
    lerna ERR! src/common/dragpanel.ts(15,8): error TS2307: Cannot find module '@phosphor/coreutils'.
    lerna ERR! src/common/dragpanel.ts(19,8): error TS2307: Cannot find module '@phosphor/dragdrop'.
```
To reproduce, try to build this `Dockerfile`:

```Dockerfile
FROM ubuntu:18.04

ENV SHELL /bin/bash

RUN apt-get update -qq \
 && DEBIAN_FRONTEND=noninteractive apt-get install -yq -qq --no-install-recommends \
   git \
   nodejs \
   npm \
   python3 \
   python3-distutils \
   python3-pip \
 && apt-get clean \
 && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*

RUN pip3 install setuptools wheel
RUN npm install -g npm
RUN npm install -g typescript

WORKDIR /tmp
RUN git clone https://github.com/jupyter/nbdime.git
WORKDIR /tmp/nbdime
RUN pip3 install .
```

**Comment by vidartf** [comment]
I'm not able to reproduce this locally (wo/docker), but I notice that even if you run `npm install -g npm`, you still see the output `lerna ERR! npm ERR! npm  v3.5.2`. I think you need at least v4 for a source install.

**Comment by vidartf** [comment]
Were you able to get this to work?

**Comment by vidartf** [comment]
Closing due to lack of activity.

**Comment by matthias-k** [comment]
I finally managed to look into this again. Your point about the npm version was correct, althouth I installed the newest version. Apparently `npm install -g npm` installs the new npm into `/usr/local/bin/npm` while there is still the old npm at `/usr/bin/npm`. Running `which npm` correcty yields the new version. Apparently the setup script here detects a wrong npm version although `$PATH` is correct: `/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin`. From a first look at `setupbase.py` I couldn't see the problem but maybe you have an idea?
